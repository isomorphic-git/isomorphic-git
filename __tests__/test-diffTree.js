/* eslint-env node, browser, jasmine */
const { makeFixture } = require('./__helpers__/FixtureFS.js')
const snapshots = require('./__snapshots__/test-diffTree.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')

const { diffTree } = require('isomorphic-git')

describe('diffTree', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })

  it('diff two empty commits', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'master', after: 'master' })
    expect(diff).toEqual({})
  })

  it('diff commit against itself', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'mainline' })
    expect(diff).toEqual({})
  })

  it('diff commit against empty commit', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'master', after: 'mainline' })
    expect(diff).toMatchSnapshot()
  })

  it('diff empty commit against commit', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'master' })
    expect(diff).toMatchSnapshot()
  })

  it('diff add-files against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'add-files' })
    expect(diff).toMatchSnapshot()
  })
  
  it('diff remove-files against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'remove-files' })
    expect(diff).toMatchSnapshot()
  })

  it('diff a-file against a-folder', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'a-folder', after: 'a-file' })
    expect(diff).toMatchSnapshot()
  })

  it('diff a-folder against a-file', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'a-file', after: 'a-folder' })
    expect(diff).toMatchSnapshot()
  })

  it('diff nest-folder against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'nest-folder' })
    expect(diff).toMatchSnapshot()
  })

  it('diff mainline against nest-folder', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'nest-folder', after: 'mainline' })
    expect(diff).toMatchSnapshot()
  })

  it('diff folder-replaces-file against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'folder-replaces-file' })
    expect(diff).toMatchSnapshot()
  })

  it('diff rename-folder against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'rename-folder' })
    expect(diff).toMatchSnapshot()
  })

  it('diff change-modes against mainline', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'mainline', after: 'change-modes' })
    expect(diff).toMatchSnapshot()
  })

  it('diff mainline against change-modes', async () => {
    // Setup
    let { gitdir } = await makeFixture('test-diff')
    // Test
    let diff = await diffTree({ gitdir, before: 'change-modes', after: 'mainline' })
    expect(diff).toMatchSnapshot()
  })

})
