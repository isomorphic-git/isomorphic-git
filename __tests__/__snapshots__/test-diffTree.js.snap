// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diffTree diff a-file against a-folder 1`] = `
Array [
  Object {
    "after": undefined,
    "before": "61780798228d17af2d34fce4cfbdf35556832472",
    "filepath": "a/b",
    "op": "rm",
  },
  Object {
    "after": "78981922613b2afb6025042ff6bd878ac1994e85",
    "before": "6be660545b31f61a82a87d2b1915f0b88bb9f16f",
    "filepath": "a",
    "op": "rmdir-write",
  },
]
`;

exports[`diffTree diff a-folder against a-file 1`] = `
Array [
  Object {
    "after": "6be660545b31f61a82a87d2b1915f0b88bb9f16f",
    "before": "78981922613b2afb6025042ff6bd878ac1994e85",
    "filepath": "a",
    "op": "rm-mkdir",
  },
  Object {
    "after": "61780798228d17af2d34fce4cfbdf35556832472",
    "before": undefined,
    "filepath": "a/b",
    "op": "write",
  },
]
`;

exports[`diffTree diff commit against empty commit 1`] = `
Array [
  Object {
    "after": "78981922613b2afb6025042ff6bd878ac1994e85",
    "before": undefined,
    "filepath": "a",
    "op": "write",
  },
  Object {
    "after": "61780798228d17af2d34fce4cfbdf35556832472",
    "before": undefined,
    "filepath": "b",
    "op": "write",
  },
  Object {
    "after": "faceb189bb665fe7a631602d904e57a4014187b7",
    "before": undefined,
    "filepath": "c",
    "op": "mkdir",
  },
  Object {
    "after": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "before": undefined,
    "filepath": "c/d",
    "op": "write",
  },
  Object {
    "after": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "before": undefined,
    "filepath": "c/e",
    "op": "write",
  },
]
`;

exports[`diffTree diff commit against itself 1`] = `Array []`;

exports[`diffTree diff empty commit against commit 1`] = `
Array [
  Object {
    "after": undefined,
    "before": "78981922613b2afb6025042ff6bd878ac1994e85",
    "filepath": "a",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "61780798228d17af2d34fce4cfbdf35556832472",
    "filepath": "b",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "filepath": "c/d",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "filepath": "c/e",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "faceb189bb665fe7a631602d904e57a4014187b7",
    "filepath": "c",
    "op": "rmdir",
  },
]
`;

exports[`diffTree diff folder-replaces-file against mainline 1`] = `
Array [
  Object {
    "after": "faceb189bb665fe7a631602d904e57a4014187b7",
    "before": "61780798228d17af2d34fce4cfbdf35556832472",
    "filepath": "b",
    "op": "rm-mkdir",
  },
  Object {
    "after": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "before": undefined,
    "filepath": "b/d",
    "op": "write",
  },
  Object {
    "after": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "before": undefined,
    "filepath": "b/e",
    "op": "write",
  },
  Object {
    "after": undefined,
    "before": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "filepath": "c/d",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "filepath": "c/e",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "faceb189bb665fe7a631602d904e57a4014187b7",
    "filepath": "c",
    "op": "rmdir",
  },
]
`;

exports[`diffTree diff mainline against nest-folder 1`] = `
Array [
  Object {
    "after": "faceb189bb665fe7a631602d904e57a4014187b7",
    "before": undefined,
    "filepath": "c",
    "op": "mkdir",
  },
  Object {
    "after": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "before": undefined,
    "filepath": "c/d",
    "op": "write",
  },
  Object {
    "after": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "before": undefined,
    "filepath": "c/e",
    "op": "write",
  },
  Object {
    "after": undefined,
    "before": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "filepath": "d/c/d",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "filepath": "d/c/e",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "faceb189bb665fe7a631602d904e57a4014187b7",
    "filepath": "d/c",
    "op": "rmdir",
  },
  Object {
    "after": undefined,
    "before": "878e0205d8bdb7d02844893647f02f49df655aff",
    "filepath": "d",
    "op": "rmdir",
  },
]
`;

exports[`diffTree diff nest-folder against mainline 1`] = `
Array [
  Object {
    "after": undefined,
    "before": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "filepath": "c/d",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "filepath": "c/e",
    "op": "rm",
  },
  Object {
    "after": undefined,
    "before": "faceb189bb665fe7a631602d904e57a4014187b7",
    "filepath": "c",
    "op": "rmdir",
  },
  Object {
    "after": "878e0205d8bdb7d02844893647f02f49df655aff",
    "before": undefined,
    "filepath": "d",
    "op": "mkdir",
  },
  Object {
    "after": "faceb189bb665fe7a631602d904e57a4014187b7",
    "before": undefined,
    "filepath": "d/c",
    "op": "mkdir",
  },
  Object {
    "after": "4bcfe98e640c8284511312660fb8709b0afa888e",
    "before": undefined,
    "filepath": "d/c/d",
    "op": "write",
  },
  Object {
    "after": "d905d9da82c97264ab6f4920e20242e088850ce9",
    "before": undefined,
    "filepath": "d/c/e",
    "op": "write",
  },
]
`;

exports[`diffTree diff two empty commits 1`] = `Array []`;
